api/src/types.rs

```diff
@@ -197,44 +254,143 @@ impl OutputPrintable {
 		let spent = chain.is_unspent(&out_id).is_err();
 
 		let proof = if include_proof {
-			Some(util::to_hex(output.proof.bytes().to_vec()))
+			Some(output.proof)
 		} else {
 			None
 		};
 
 		OutputPrintable {
-			output_type: output_type,
-			commit: util::to_hex(output.commit.0.to_vec()),
-			switch_commit_hash: output.switch_commit_hash.to_hex(),
-			spent: spent,
-			proof: proof,
+			output_type,
+			commit: output.commit,
+			switch_commit_hash: output.switch_commit_hash,
+			spent,
+			proof,
 			proof_hash: util::to_hex(output.proof.hash().to_vec()),
 		}
 	}
 
 	// Convert the hex string back into a switch_commit_hash instance
 	pub fn switch_commit_hash(&self) -> Result<core::SwitchCommitHash, ser::Error> {
-		core::SwitchCommitHash::from_hex(&self.switch_commit_hash)
+		Ok(self.switch_commit_hash.clone())
 	}
 
 	pub fn commit(&self) -> Result<pedersen::Commitment, ser::Error> {
-		let vec = util::from_hex(self.commit.clone())
-			.map_err(|_| ser::Error::HexError(format!("output commit hex_error")))?;
-		Ok(pedersen::Commitment::from_vec(vec))
+		Ok(self.commit.clone())
 	}
 
 	pub fn range_proof(&self) -> Result<pedersen::RangeProof, ser::Error> {
-		if let Some(ref proof) = self.proof {
-			let vec = util::from_hex(proof.clone())
-				.map_err(|_| ser::Error::HexError(format!("output range_proof hex_error")))?;
-			let mut bytes = [0; MAX_PROOF_SIZE];
-			for i in 0..vec.len() {
-				bytes[i] = vec[i];
+		self.proof.clone().ok_or_else(|| ser::Error::HexError(format!("output range_proof missing")))
+	}
+}
+
+impl serde::ser::Serialize for OutputPrintable {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error> where
+		S: serde::ser::Serializer {
+		let mut state = serializer.serialize_struct("OutputPrintable", 6)?;
+		state.serialize_field("output_type", &self.output_type)?;
+		state.serialize_field("commit", &util::to_hex(self.commit.0.to_vec()))?;
+		state.serialize_field("switch_commit_hash", &self.switch_commit_hash.to_hex())?;
+		state.serialize_field("spent", &self.spent)?;
+		state.serialize_field("proof", &self.proof)?;
+		state.serialize_field("proof_hash", &self.proof_hash)?;
+		state.end()
+	}
+}
+
+impl<'de> serde::de::Deserialize<'de> for OutputPrintable {
```

Comment:
I guess this is the price we pay for telling serde to do the deserialization for us...